pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -666,61 +711,87 @@ func (r *ReconcileLogStorage) getElasticsearchCertificateSecrets(ctx context.Con
+		if err != nil {
+			return nil, nil, nil, nil, err
+		}
+
+		if elasticsearchPubSecret != nil {
+			// If the provided certificate secret (secret) is managed by the operator we need to check if the secret that
+			// Elasticsearch creates from that given secret (pubSecret) has the expected DNS name. If it doesn't, delete the
+			// public secret so it can get recreated.
+			if !customerProvidedCert {
+				err = utils.SecretHasExpectedDNSNames(elasticsearchPubSecret, corev1.TLSCertKey, svcDNSNames)
+				if err == utils.ErrInvalidCertDNSNames {
+					if err := r.deleteInvalidECKManagedPublicCertSecret(ctx, elasticsearchPubSecret); err != nil {
+						return nil, nil, nil, nil, err
+					}
+				}
+			}
+
+			elasticsearchCertSecret = rsecret.CopyToNamespace(rmeta.OperatorNamespace(), elasticsearchPubSecret)[0]
+		} else {
+			elasticsearchCertSecret = render.CreateCertificateSecret(oprKeyCert.Data[corev1.TLSCertKey], relasticsearch.PublicCertSecret, rmeta.OperatorNamespace())
+		}
+
+		// Get the kibana pub secret - might be nil
+		kibanaPubSecret, err := utils.GetSecret(ctx, r.client, render.KibanaPublicCertSecret, render.KibanaNamespace)
 		if err != nil {
-			return nil, nil, nil, err
+			return nil, nil, nil, nil, err
 		}
 
-		if pubSecret != nil {
+		if kibanaPubSecret != nil {
 			// If the provided certificate secret (secret) is managed by the operator we need to check if the secret that
 			// Elasticsearch creates from that given secret (pubSecret) has the expected DNS name. If it doesn't, delete the
 			// public secret so it can get recreated.
 			if !customerProvidedCert {
-				err = utils.SecretHasExpectedDNSNames(pubSecret, corev1.TLSCertKey, svcDNSNames)
+				err = utils.SecretHasExpectedDNSNames(kibanaPubSecret, corev1.TLSCertKey, svcDNSNames)
 				if err == utils.ErrInvalidCertDNSNames {
-					if err := r.deleteInvalidECKManagedPublicCertSecret(ctx, pubSecret); err != nil {
-						return nil, nil, nil, err
+					if err := r.deleteInvalidECKManagedPublicCertSecret(ctx, kibanaPubSecret); err != nil {
+						return nil, nil, nil, nil, err
 					}
 				}
 			}
 
-			certSecret = rsecret.CopyToNamespace(rmeta.OperatorNamespace(), pubSecret)[0]
+			kibanaCertSecret = rsecret.CopyToNamespace(rmeta.OperatorNamespace(), kibanaPubSecret)[0]
+		} else {
+			kibanaCertSecret = render.CreateCertificateSecret(oprKeyCert.Data[corev1.TLSCertKey], render.KibanaPublicCertSecret, rmeta.OperatorNamespace())
 		}
 	}
 
-	return oprKeyCert, esKeyCert, certSecret, err
+	return oprKeyCert, esKeyCert, elasticsearchCertSecret, kibanaCertSecret, err
 }
 
 func (r *ReconcileLogStorage) kibanaSecrets(ctx context.Context, instl *operatorv1.InstallationSpec) ([]*corev1.Secret, error) {
 
 	var secrets []*corev1.Secret
-	svcDNSNames := dns.GetServiceDNSNames(render.KibanaServiceName, render.KibanaNamespace, r.clusterDomain)
+	svcDNSNames := dns.GetServiceDNSNames(render.KibanaInternalServiceName, render.KibanaNamespace, r.clusterDomain)
```

Comment:
I think you should adjust the name of this function if it is suppose to be handling "Internal" secret configuration.